src/resolvers/__tests__/resolvers.test.js

```diff
@@ -70,6 +70,47 @@ test('AssetResolver.collect returns empty object when file not in the list', ()
   expect(result).toEqual({});
 });
 
+test('AssetResolver.collect returns files based on requested platform', () => {
+  const fileList = [
+    'mail.android.png',
+    'mail.png',
+    'mail@2x.android.png',
+    'mail@2x.png',
+    'mail@3x.android.png',
+    'mail@3x.png',
+  ];
+  const fileListDifferentOrder = [
+    'mail.png',
+    'mail.android.png',
+    'mail@2x.png',
+    'mail@2x.android.png',
+    'mail@3x.png',
+    'mail@3x.android.png',
```

Comment:
can you add more platforms? some `.native.png` and `.ios.png` files to make sure this works correctly